gubernator/main.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import functools
+import json
+import re
+import os
+import urllib
+import zlib
+
+import webapp2
+import jinja2
+
+from google.appengine.api import memcache
+
+import defusedxml.ElementTree as ET
+import cloudstorage as gcs
+
+import filters
+
+BUCKET_WHITELIST = {
+    'kubernetes-jenkins',
+}
+
+DEFAULT_JOBS = {
+    'kubernetes-jenkins/logs/': {
+        'kubelet-gce-e2e-ci',
+        'kubernetes-build',
+        'kubernetes-e2e-gce',
+        'kubernetes-e2e-gce-scalability',
+        'kubernetes-e2e-gce-slow',
+        'kubernetes-e2e-gke',
+        'kubernetes-e2e-gke-slow',
+        'kubernetes-kubemark-5-gce',
+        'kubernetes-kubemark-500-gce',
+        'kubernetes-test-go',
+    }
+}
+
+JINJA_ENVIRONMENT = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(os.path.dirname(__file__) + '/templates'),
+    extensions=['jinja2.ext.autoescape'],
+    trim_blocks=True,
+    autoescape=True)
+JINJA_ENVIRONMENT.line_statement_prefix = '%'
+filters.register(JINJA_ENVIRONMENT.filters)
+
+
+def pad_numbers(s):
+    """Modify a string to make its numbers suitable for natural sorting."""
+    return re.sub(r'\d+', lambda m: m.group(0).rjust(16, '0'), s)
+
+
+def memcache_memoize(prefix, exp=60 * 60, neg_exp=60):
```

Comment:
nit: I would prefer `expires` instead of `exp`, since the latter means exponential in my experience.